website/src/pages/tutorial/basic/08-graph-relations.mdx

```diff
@@ -392,22 +392,42 @@ type Link {
   id: ID!
   description: String!
   url: String!
-  comments: [Comment!]
+  comments: [Comment!]!
 }
 ```
 
 On the Prisma API level, the method call for loading will always return an array. It does not
 distinguish between an empty array and an array with comments.
 
-On the GraphQL schema, we could follow the same practice, and make the comments array non-nullable
-(`[Comment!]!`).
+On the GraphQL schema, we have three options:
 
-In our example, we've choosen the nullable option, as it indicates the API the need to handle that
-case, when they will generate the types on their end.
+1. `comments: [Comment]`
+2. `comments: [Comment!]`
+3. `comments: [Comment!]!`
```

Comment:
We even have a fourth option: `comments: [Comment]!`. I don't know if it's worth explaining this one too ?

Reply:
what would you write about that?
can you add a text similar to that?

Reply:
> Option 4, `[Comment]!`, shares semantics with option 1 and 3. Like 1 the array elements can be  `null` or `Comment`, but like 3, the field itself will _always_ be an array, never `null`. This type formulation is rarely used because simply omitting a value from the array is typically sufficient to indicate its non-valueless.

Reply:
I would maybe alter the documentation here to talk about two aspects which cut across all four combinations:

1. nullable array elements
2. nullable array

After talking about those two aspects, one can simply reference which aspects apply to which combinations.

Then, a final word about which formulation is the most popular and why (`[X!]!` because it provides the simplest way to sufficiently model empty states blah blah).

Reply:
shit I missed your comment!
can you create a short PR just with some words?
I can then take the time to turn it into something readable